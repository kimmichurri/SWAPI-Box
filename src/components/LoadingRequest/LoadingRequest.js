import React from 'react';
import classic_loading_icon from '../../assets/classic_loading_icon.png';

export default function LoadingRequest() {
    return (
    <div className="loading-container">
        <img
            className="loading-icon"
            src={classic_loading_icon}
            alt="loading icon, the page is loading"
        />
        <p className="loading-text">Loading...</p>
    </div>
    )
}