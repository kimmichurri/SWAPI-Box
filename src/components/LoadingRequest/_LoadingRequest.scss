.loading-text {
    color: white;
    font-family: $main-font;
    font-size: 20px;
}

.loading-icon {
    animation: spin 2s linear infinite;
    width: 50px;
    margin-top: 30px;
}

@keyframes spin {
    0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}