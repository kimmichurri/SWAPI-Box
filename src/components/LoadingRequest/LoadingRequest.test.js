import React from 'react';
import { shallow } from 'enzyme';
import LoadingRequest from './LoadingRequest'

describe('LoadingRequest', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<LoadingRequest />)
  })

  it('should match the snapshot', () => {
    expect(wrapper.debug()).toMatchSnapshot()
  })
})