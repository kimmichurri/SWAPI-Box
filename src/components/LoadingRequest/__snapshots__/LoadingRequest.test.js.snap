// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingRequest should match the snapshot 1`] = `
"<div className=\\"loading-container\\">
  <img className=\\"loading-icon\\" src=\\"classic_loading_icon.png\\" alt=\\"loading icon, the page is loading\\" />
  <p className=\\"loading-text\\">
    Loading...
  </p>
</div>"
`;
